59-bottlenose-dolphin/technical_specification.md

```diff
@@ -0,0 +1,111 @@
+  - Settle on a component library. See our earlier evaluation [here](https://github.com/ghga-de/adrs/pull/9). Did we miss anything, is there anything new?
+  - How important are responsive design and accessibility to us? How well is this supported by the chosen system/library?
+  - Do we want to use plain CSS or SASS or SCSS syntax?
+
+- State management and other tooling:
+  - Shall we use a dedicated state management solution (like NgRx, Akita, Elf or NgxZustand)? Or start with a NGRX signal store or simple custom services to manage state?
+  - Should we primarily use RxJS or Signals?
+  - Shall we make use of a caching library for HTTP requests like NgHttpCaching or cashew?
+  - Are there any client libraries or development tools that we should make use of?
+
+- Deployment and configuration:
+  - How can we pass a runtime configuration similarly as we do with our microservices?
+  - How can we inject the configuration into the application without rebuilding the complete application, as we currently do for the React app?
+  - How can we rewrite the build and run steps from Python to JavaScript so that we do not need to install Python in the Docker container anymore?
+  - What is the best way to serve the SPA in production? We currently use the "serve" package from npm. Is that still up to date and recommended?
+  - How can we build a very slim Docker image for production that does not contain any build or development tools and other unnecessary components?
+  - Which JavaScript runtime shall we use (Node with yarn or npm, Deno, Bun)?
+
+- Searchability and performance:
+  - Define a suitable HTML structure using semantic HTML.
+  - Shall we support other semantic web technologies like schema.org with JSON-LD?
+  - Can we make use of [deferrable views](https://angular.dev/guide/defer)?
+  - Is it worthwhile for us to support server-side rendering?
+  - What is the current state of server-side-rendering in Angular? How much effort is it to make use of it?
+
+- Testing:
+  - How can we (manually) test against the real backend? What would be a replacement for `setupProxy.js`?
+  - How can we (manually) test against a mock backend? Shall we continue to use [MSW](https://mswjs.io/) with simple, ideally static responses?
+  - Do we want to implement both unit tests and end-2-end-tests and in which proportions? Which degree of coverage do we want to achieve?
+  - Which unit testing tool and test runner should we use?
+  - Which end-to-end-testing tool (e.g. Cypress, Nightwatch, WebdriverIO, Puppeteer, Playwright)?
+  - Do these tools work well with MSW? Can we run them as GitHub actions?
+
+- Migration:
+  - How many developer resources should we put into the re-writing? How quickly do we want to finish this?
+  - How do we deal with new feature request before the migration is finished?
+  - Which features should be implemented first?
+
```

Comment:
- Documentation: 
    - How do we document the new codebase?
    - Do we set any documentation requirements?
    - How do we onboard others to the new codebase?
    - Should we document UI components with an abstraction layer (storybook.js)

Reply:
added this in 75f27146